pkg/ibmvpc/nacl_analysis.go

```diff
@@ -445,6 +462,76 @@ func (na *NACLAnalyzer) AllowedConnectivity(subnetCidr, inSubentCidr, target str
 		}
 	}
 	// expecting disjoint ip-blocks, thus not expecting to get here
-	return nil, fmt.Errorf("isIngress: %t , target %s, subnetCidr: %s, inSubentCidr %s,  "+
-		"could not find allowed connectivity for given target + inSubentCidr", isIngress, target, subnetCidr, inSubentCidr)
+	return nil, fmt.Errorf(notFoundMsg, isIngress, target, subnetCidr, inSubentCidr)
+}
+
+// rulesInConnectivity returns set of rules contributing to a connections given src/dst and direction
+// if conn is specified then rules contributing to that connection; otherwise to any connection src->dst
+// if the input subnet was not yet analyzed, it first adds its analysis to saved results
+func (na *NACLAnalyzer) rulesInConnectivity(subnetCidr, inSubentCidr,
+	target string, conn *common.ConnectionSet, isIngress bool) ([]int, error) {
+	// add analysis of the given subnet
+	// analyzes per subnet disjoint cidrs (it is not necessarily entire subnet cidr)
+	analyzedConns, targetIPblock, inSubnetIPblock := na.initConnectivityRelatedCompute(subnetCidr, inSubentCidr, target, isIngress)
+
+	for disjointSubnetCidr, analyzedConnsPerCidr := range analyzedConns {
+		disjointSubnetCidrIPblock, err := common.IPBlockFromIPRangeStr(disjointSubnetCidr)
+		if err != nil {
+			return nil, err
+		}
+		if inSubnetIPblock.ContainedIn(disjointSubnetCidrIPblock) {
+			for resTarget, rules := range analyzedConnsPerCidr.contribRules {
+				if targetIPblock.ContainedIn(resTarget) {
```

Comment:
Makes sense, only instead of
```
if conn != nil { 
    					    return rules, nil
					} 
```
it should be
```
if conn == nil { 
    					    return rules, nil
					}
```
I renamed conn to connQuery so that it will be clear